commons/src/test/java/io/hyscale/commons/utils/LogProcessorTests.java

```diff
@@ -0,0 +1,181 @@
+import io.hyscale.commons.handler.SampleLogHandler;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class LogProcessorTests {
+    private static String logFileContent = "logger running" + "\n" + "logger Running";
+    private LogProcessor logProcessor = new LogProcessor();
+    private static String logFilePath = "/tmp/logs/logs.txt";
+    private static String encoding = "UTF-8";
+    private static File file;
+    private SampleLogHandler sampleLogHandler = new SampleLogHandler();
+
+
+    @BeforeEach
+    public void getLogFile() {
+        file = FileUtils.getFile(logFilePath);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+        }
+    }
+
+    public static Stream<Arguments> getNullInputsForWrite() {
+        InputStream inputStream = getInputStream();
+        return Stream.of(Arguments.of(inputStream, null),
+                Arguments.of(null, logFilePath),
+                Arguments.of(inputStream, ""));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "getNullInputsForWrite")
+    public void testNullConditionsForWrite(InputStream inputStream, String logFilePath) {
+        Assertions.assertThrows(HyscaleException.class, () -> {
+            logProcessor.writeLogFile(inputStream, logFilePath);
+        });
+    }
+
+    @Test
+    public void testWriteLogFile() {
+        String content = null;
+        InputStream inputStream = getInputStream();
+        try {
+            logProcessor.writeLogFile(inputStream, logFilePath);
+            content = FileUtils.readFileToString(file, encoding).trim();
+            inputStream.close();
+        } catch (IOException i) {
+            Assertions.fail();
+        } catch (HyscaleException e) {
+        }
+        Assertions.assertNotNull(content);
+        Assertions.assertEquals(content, logFileContent);
+    }
+
+    public static Stream<Arguments> getNullInputsForRead() {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        return Stream.of(Arguments.of(null, os, null),
+                Arguments.of(file, null, null),
+                Arguments.of(file, os, 1),
+                Arguments.of(file, os, null));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "getNullInputsForRead")
+    public void readLogFileTest(File readFile, OutputStream os, Integer lines) {
+        if (readFile == null || !readFile.isDirectory() || !readFile.exists()) {
+        } else if (os == null) {
+            Assertions.assertThrows(HyscaleException.class, () -> {
+                logProcessor.readLogFile(readFile, os, lines);
+            });
+        } else {
```

Comment:
Keep the positive testcase separate, don't mix the test cases. We will loose the readability of the test case.